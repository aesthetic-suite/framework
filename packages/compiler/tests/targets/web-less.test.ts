import { runTargetTests } from './runTargetTests';

runTargetTests('web', 'web-less');
