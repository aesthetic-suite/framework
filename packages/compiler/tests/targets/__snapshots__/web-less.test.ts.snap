// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Format web-less compiles and writes files 1`] = `
"// Auto-generated by Aesthetic. Do not modify!

@border-sm-radius: 0.14rem; // 2px
@border-sm-width: 0.04rem; // 0.50px
@border-df-radius: 0.21rem; // 3px
@border-df-width: 0.07rem; // 1px
@border-lg-radius: 0.32rem; // 4.50px
@border-lg-width: 0.14rem; // 2px

@breakpoint-xs-query: 'screen and (min-width: 45.71em)';
@breakpoint-xs-query-size: 640;
@breakpoint-xs-root-line-height: 1.33;
@breakpoint-xs-root-text-size: 14.94px;
@breakpoint-sm-query: 'screen and (min-width: 68.57em)';
@breakpoint-sm-query-size: 960;
@breakpoint-sm-root-line-height: 1.42;
@breakpoint-sm-root-text-size: 15.94px;
@breakpoint-md-query: 'screen and (min-width: 91.43em)';
@breakpoint-md-query-size: 1280;
@breakpoint-md-root-line-height: 1.52;
@breakpoint-md-root-text-size: 17.01px;
@breakpoint-lg-query: 'screen and (min-width: 114.29em)';
@breakpoint-lg-query-size: 1600;
@breakpoint-lg-root-line-height: 1.62;
@breakpoint-lg-root-text-size: 18.15px;
@breakpoint-xl-query: 'screen and (min-width: 137.14em)';
@breakpoint-xl-query-size: 1920;
@breakpoint-xl-root-line-height: 1.73;
@breakpoint-xl-root-text-size: 19.36px;

@depth-content: 100;
@depth-navigation: 1000;
@depth-sheet: 1100;
@depth-overlay: 1200;
@depth-modal: 1300;
@depth-toast: 1400;
@depth-dialog: 1500;
@depth-menu: 1600;
@depth-tooltip: 1700;

@heading-l1-letter-spacing: 0.25px;
@heading-l1-line-height: 1.5;
@heading-l1-size: 1.14rem; // 16px
@heading-l2-letter-spacing: 0.33px;
@heading-l2-line-height: 1.69;
@heading-l2-size: 1.43rem; // 20px
@heading-l3-letter-spacing: 0.44px;
@heading-l3-line-height: 1.9;
@heading-l3-size: 1.79rem; // 25px
@heading-l4-letter-spacing: 0.59px;
@heading-l4-line-height: 2.14;
@heading-l4-size: 2.23rem; // 31.25px
@heading-l5-letter-spacing: 0.79px;
@heading-l5-line-height: 2.4;
@heading-l5-size: 2.79rem; // 39.06px
@heading-l6-letter-spacing: 1.05px;
@heading-l6-line-height: 2.7;
@heading-l6-size: 3.49rem; // 48.83px

@shadow-xs-x: 0rem; // 0px
@shadow-xs-y: 0.14rem; // 2px
@shadow-xs-blur: 0.14rem; // 2px
@shadow-xs-spread: 0rem; // 0px
@shadow-sm-x: 0rem; // 0px
@shadow-sm-y: 0.23rem; // 3.24px
@shadow-sm-blur: 0.25rem; // 3.50px
@shadow-sm-spread: 0rem; // 0px
@shadow-md-x: 0rem; // 0px
@shadow-md-y: 0.37rem; // 5.24px
@shadow-md-blur: 0.44rem; // 6.13px
@shadow-md-spread: 0rem; // 0px
@shadow-lg-x: 0rem; // 0px
@shadow-lg-y: 0.61rem; // 8.47px
@shadow-lg-blur: 0.77rem; // 10.72px
@shadow-lg-spread: 0rem; // 0px
@shadow-xl-x: 0rem; // 0px
@shadow-xl-y: 0.98rem; // 13.71px
@shadow-xl-blur: 1.34rem; // 18.76px
@shadow-xl-spread: 0rem; // 0px

@spacing-xs: 0.31rem; // 4.38px
@spacing-sm: 0.63rem; // 8.75px
@spacing-df: 1.25rem; // 17.50px
@spacing-md: 2.5rem; // 35px
@spacing-lg: 3.75rem; // 52.50px
@spacing-xl: 5rem; // 70px
@spacing-unit: 17.5;

@text-sm-line-height: 1.25;
@text-sm-size: 0.89rem; // 12.44px
@text-df-line-height: 1.25;
@text-df-size: 1rem; // 14px
@text-lg-line-height: 1.25;
@text-lg-size: 1.13rem; // 15.75px

@typography-font-heading: 'system-ui, -apple-system, BlinkMacSystemFont, \\"Segoe UI\\", Roboto, \\"Helvetica Neue\\", Arial, \\"Noto Sans\\", sans-serif, \\"Apple Color Emoji\\", \\"Segoe UI Emoji\\", \\"Segoe UI Symbol\\", \\"Noto Color Emoji\\"';
@typography-font-monospace: 'Menlo, Monaco, Consolas, \\"Lucida Console\\", \\"Liberation Mono\\", \\"Courier New\\", monospace';
@typography-font-text: 'system-ui, -apple-system, BlinkMacSystemFont, \\"Segoe UI\\", Roboto, \\"Helvetica Neue\\", Arial, \\"Noto Sans\\", sans-serif, \\"Apple Color Emoji\\", \\"Segoe UI Emoji\\", \\"Segoe UI Symbol\\", \\"Noto Color Emoji\\"';
@typography-font-system: 'system-ui, -apple-system, BlinkMacSystemFont, \\"Segoe UI\\", Roboto, \\"Helvetica Neue\\", Arial, \\"Noto Sans\\", sans-serif, \\"Apple Color Emoji\\", \\"Segoe UI Emoji\\", \\"Segoe UI Symbol\\", \\"Noto Color Emoji\\"';
@typography-root-line-height: 1.25;
@typography-root-text-size: 14px;
"
`;

exports[`Format web-less compiles and writes files 2`] = `
"// Auto-generated by Aesthetic. Do not modify!
@import '..';

// contrast: normal
// scheme: light

@palette-brand-color-00: #000;
@palette-brand-color-10: #000;
@palette-brand-color-20: #000;
@palette-brand-color-30: #000;
@palette-brand-color-40: #000;
@palette-brand-color-50: #000;
@palette-brand-color-60: #000;
@palette-brand-color-70: #000;
@palette-brand-color-80: #000;
@palette-brand-color-90: #000;
@palette-brand-bg-base: #000;
@palette-brand-bg-focused: #000;
@palette-brand-bg-hovered: #000;
@palette-brand-bg-selected: #000;
@palette-brand-bg-disabled: #000;
@palette-brand-fg-base: #000;
@palette-brand-fg-focused: #000;
@palette-brand-fg-hovered: #000;
@palette-brand-fg-selected: #000;
@palette-brand-fg-disabled: #000;

@palette-primary-color-00: #000;
@palette-primary-color-10: #000;
@palette-primary-color-20: #000;
@palette-primary-color-30: #000;
@palette-primary-color-40: #000;
@palette-primary-color-50: #000;
@palette-primary-color-60: #000;
@palette-primary-color-70: #000;
@palette-primary-color-80: #000;
@palette-primary-color-90: #000;
@palette-primary-bg-base: #000;
@palette-primary-bg-focused: #000;
@palette-primary-bg-hovered: #000;
@palette-primary-bg-selected: #000;
@palette-primary-bg-disabled: #000;
@palette-primary-fg-base: #000;
@palette-primary-fg-focused: #000;
@palette-primary-fg-hovered: #000;
@palette-primary-fg-selected: #000;
@palette-primary-fg-disabled: #000;

@palette-secondary-color-00: #000;
@palette-secondary-color-10: #000;
@palette-secondary-color-20: #000;
@palette-secondary-color-30: #000;
@palette-secondary-color-40: #000;
@palette-secondary-color-50: #000;
@palette-secondary-color-60: #000;
@palette-secondary-color-70: #000;
@palette-secondary-color-80: #000;
@palette-secondary-color-90: #000;
@palette-secondary-bg-base: #000;
@palette-secondary-bg-focused: #000;
@palette-secondary-bg-hovered: #000;
@palette-secondary-bg-selected: #000;
@palette-secondary-bg-disabled: #000;
@palette-secondary-fg-base: #000;
@palette-secondary-fg-focused: #000;
@palette-secondary-fg-hovered: #000;
@palette-secondary-fg-selected: #000;
@palette-secondary-fg-disabled: #000;

@palette-tertiary-color-00: #000;
@palette-tertiary-color-10: #000;
@palette-tertiary-color-20: #000;
@palette-tertiary-color-30: #000;
@palette-tertiary-color-40: #000;
@palette-tertiary-color-50: #000;
@palette-tertiary-color-60: #000;
@palette-tertiary-color-70: #000;
@palette-tertiary-color-80: #000;
@palette-tertiary-color-90: #000;
@palette-tertiary-bg-base: #000;
@palette-tertiary-bg-focused: #000;
@palette-tertiary-bg-hovered: #000;
@palette-tertiary-bg-selected: #000;
@palette-tertiary-bg-disabled: #000;
@palette-tertiary-fg-base: #000;
@palette-tertiary-fg-focused: #000;
@palette-tertiary-fg-hovered: #000;
@palette-tertiary-fg-selected: #000;
@palette-tertiary-fg-disabled: #000;

@palette-neutral-color-00: #000;
@palette-neutral-color-10: #000;
@palette-neutral-color-20: #000;
@palette-neutral-color-30: #000;
@palette-neutral-color-40: #000;
@palette-neutral-color-50: #000;
@palette-neutral-color-60: #000;
@palette-neutral-color-70: #000;
@palette-neutral-color-80: #000;
@palette-neutral-color-90: #000;
@palette-neutral-bg-base: #000;
@palette-neutral-bg-focused: #000;
@palette-neutral-bg-hovered: #000;
@palette-neutral-bg-selected: #000;
@palette-neutral-bg-disabled: #000;
@palette-neutral-fg-base: #000;
@palette-neutral-fg-focused: #000;
@palette-neutral-fg-hovered: #000;
@palette-neutral-fg-selected: #000;
@palette-neutral-fg-disabled: #000;

@palette-muted-color-00: #000;
@palette-muted-color-10: #000;
@palette-muted-color-20: #000;
@palette-muted-color-30: #000;
@palette-muted-color-40: #000;
@palette-muted-color-50: #000;
@palette-muted-color-60: #000;
@palette-muted-color-70: #000;
@palette-muted-color-80: #000;
@palette-muted-color-90: #000;
@palette-muted-bg-base: #000;
@palette-muted-bg-focused: #000;
@palette-muted-bg-hovered: #000;
@palette-muted-bg-selected: #000;
@palette-muted-bg-disabled: #000;
@palette-muted-fg-base: #000;
@palette-muted-fg-focused: #000;
@palette-muted-fg-hovered: #000;
@palette-muted-fg-selected: #000;
@palette-muted-fg-disabled: #000;

@palette-danger-color-00: #000;
@palette-danger-color-10: #000;
@palette-danger-color-20: #000;
@palette-danger-color-30: #000;
@palette-danger-color-40: #000;
@palette-danger-color-50: #000;
@palette-danger-color-60: #000;
@palette-danger-color-70: #000;
@palette-danger-color-80: #000;
@palette-danger-color-90: #000;
@palette-danger-bg-base: #000;
@palette-danger-bg-focused: #000;
@palette-danger-bg-hovered: #000;
@palette-danger-bg-selected: #000;
@palette-danger-bg-disabled: #000;
@palette-danger-fg-base: #000;
@palette-danger-fg-focused: #000;
@palette-danger-fg-hovered: #000;
@palette-danger-fg-selected: #000;
@palette-danger-fg-disabled: #000;

@palette-warning-color-00: #000;
@palette-warning-color-10: #000;
@palette-warning-color-20: #000;
@palette-warning-color-30: #000;
@palette-warning-color-40: #000;
@palette-warning-color-50: #000;
@palette-warning-color-60: #000;
@palette-warning-color-70: #000;
@palette-warning-color-80: #000;
@palette-warning-color-90: #000;
@palette-warning-bg-base: #000;
@palette-warning-bg-focused: #000;
@palette-warning-bg-hovered: #000;
@palette-warning-bg-selected: #000;
@palette-warning-bg-disabled: #000;
@palette-warning-fg-base: #000;
@palette-warning-fg-focused: #000;
@palette-warning-fg-hovered: #000;
@palette-warning-fg-selected: #000;
@palette-warning-fg-disabled: #000;

@palette-failure-color-00: #000;
@palette-failure-color-10: #000;
@palette-failure-color-20: #000;
@palette-failure-color-30: #000;
@palette-failure-color-40: #000;
@palette-failure-color-50: #000;
@palette-failure-color-60: #000;
@palette-failure-color-70: #000;
@palette-failure-color-80: #000;
@palette-failure-color-90: #000;
@palette-failure-bg-base: #000;
@palette-failure-bg-focused: #000;
@palette-failure-bg-hovered: #000;
@palette-failure-bg-selected: #000;
@palette-failure-bg-disabled: #000;
@palette-failure-fg-base: #000;
@palette-failure-fg-focused: #000;
@palette-failure-fg-hovered: #000;
@palette-failure-fg-selected: #000;
@palette-failure-fg-disabled: #000;

@palette-success-color-00: #000;
@palette-success-color-10: #000;
@palette-success-color-20: #000;
@palette-success-color-30: #000;
@palette-success-color-40: #000;
@palette-success-color-50: #000;
@palette-success-color-60: #000;
@palette-success-color-70: #000;
@palette-success-color-80: #000;
@palette-success-color-90: #000;
@palette-success-bg-base: #000;
@palette-success-bg-focused: #000;
@palette-success-bg-hovered: #000;
@palette-success-bg-selected: #000;
@palette-success-bg-disabled: #000;
@palette-success-fg-base: #000;
@palette-success-fg-focused: #000;
@palette-success-fg-hovered: #000;
@palette-success-fg-selected: #000;
@palette-success-fg-disabled: #000;

@palette-info-color-00: #000;
@palette-info-color-10: #000;
@palette-info-color-20: #000;
@palette-info-color-30: #000;
@palette-info-color-40: #000;
@palette-info-color-50: #000;
@palette-info-color-60: #000;
@palette-info-color-70: #000;
@palette-info-color-80: #000;
@palette-info-color-90: #000;
@palette-info-bg-base: #000;
@palette-info-bg-focused: #000;
@palette-info-bg-hovered: #000;
@palette-info-bg-selected: #000;
@palette-info-bg-disabled: #000;
@palette-info-fg-base: #000;
@palette-info-fg-focused: #000;
@palette-info-fg-hovered: #000;
@palette-info-fg-selected: #000;
@palette-info-fg-disabled: #000;
"
`;
